### PR TITLE
Add no-optional flag to npm

### LINKED ISSUE #1421: Add --no-optional to the npm plugin
##### SUMMARY

community.general.npm should support the `--no-optional` argument.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

community.general.npm

##### ADDITIONAL INFORMATION

The `npm` plugin should support the `--no-optional` argument of the `npm` command. It is helpful in solving certain issues with package installation.

For example, using the `npm` module to install `statsd` (at least when installing it globally), it pulls in `modern-syslog` as an optional dependency. That causes `node-gyp rebuild` to be run, and that in turn exits with 1 for an unknown reason, causing the `npm` module to fail during a playbook run. In reality, the actual `statsd` package has been successfully installed. (This is on an Ubuntu 20.04 system.)

I don't have the time nor the expertise to debug why `node-gyp` is causing issues, but I could circumvent it cleanly if `--no-optional` was supported.

### PR DESCRIPTION
**Pull request by aminvakil**
##### SUMMARY
Fixes #1421 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
npm

### CODE REVIEW

**Comment by felixfontein** [comment]
You need to add the option to the argument_spec itself :)

File: plugins/modules/packaging/language/npm.py [hunk 1]
```diff
@@ -263,6 +272,7 @@ def main():
     ignore_scripts = module.params['ignore_scripts']
     unsafe_perm = module.params['unsafe_perm']
     ci = module.params['ci']
+    no_optional = module.params['no_optional']
```
Comment (felixfontein) [inline plugins/modules/packaging/language/npm.py:275]: You need to pass this to the `Npm` class.

Comment (aminvakil) [reply plugins/modules/packaging/language/npm.py:276]: Fixed in https://github.com/ansible-collections/community.general/pull/1428/commits/7d495beb0c8683191e64c9a6647f03232d07acfd.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@aminkavil thanks for implementing this!

**Pull request merged**